### PR TITLE
Update xpi-manifest version format to work with MV3

### LINKED ISSUE #185: addon+version uniqueness won't be compatible with MV3 (new version format)
The addon+version uniqueness implemented in https://github.com/mozilla-extensions/xpi-manifest/pull/135 will not be compatible with MV3 because we encourage a new simpler version string format in the manifest. In practice, it means:

- letters are no longer allowed
- the string should have between 1 and 4 numbers separated with dots
- each number should have up to 9 digits
- for non-zero numbers, leading zeros are not allowed

We'll need to update the `build` step to account for that when MV3 extensions will start to be signed. At the moment, MV3 extensions won't be accepted by the linter, though (the linter task is blocking for privileged add-ons since #169).

### PR DESCRIPTION
**Pull request by gabrielBusta**
### Description

Update `xpi-manifest` so the version format generated by the addons-pipeline is compatible with web-extension manifest MV3. Check the version for the rules outlined on issue #185. We will need to update `xpi-template` as well.

### Reference

Fixes #185 
[Bug 1824240](https://bugzilla.mozilla.org/show_bug.cgi?id=1824240)

### CODE REVIEW

File: taskcluster/docker/node/build.py [hunk 1]
```diff
@@ -127,7 +144,7 @@ def mkdir(path):
 def get_hash(path, hash_alg="sha256"):
     h = hashlib.new(hash_alg)
     with open(path, "rb") as fh:
-        for chunk in iter(functools.partial(fh.read, 4096), b''):
+        for chunk in iter(functools.partial(fh.read, 4096), b""):
```
Comment (gabrielBusta) [inline taskcluster/docker/node/build.py:147]: The changes below this line are just formatting changes caused by running `black` on this file.

**Comment by willdurand** [comment]
@gabrielBusta hey, any progress on this PR?

**Comment by gabrielBusta** [comment]
I've got this in the backlog @willdurand

I'll get some cycles soon.

Anything timing related I should be aware of?

**Comment by willdurand** [comment]
> Anything timing related I should be aware of?

Nope, this is just the last issue to address for the (new) version format.

**Comment by gabrielBusta** [comment]
There's some new check on branches names, and the + is breaking the decisions

**Pull request closed without merging**